### PR TITLE
Add support for symlinks

### PR DESCRIPTION
**Pull request by boreq**

### CODE REVIEW

**Review by coderobe — APPROVED** [review]
WFM

File: adapters/music/scanner/scanner.go [hunk 1]
```diff
@@ -153,22 +154,31 @@ func (s *Scanner) Start() (<-chan Album, error) {
...
 		}
 
+		realPath, err := filepath.EvalSymlinks(path)
+		if err != nil {
+			return errors.Wrap(err, "could not eval a symlink")
+		}
+
+		_, ok := visited[realPath]
+		if ok {
+			return fmt.Errorf("look detected: '%s' visited multiple times", realPath)
```
Comment (coderobe) [inline adapters/music/scanner/scanner.go:172]: typo: `look` -> `loop`?

Comment (boreq) [inline adapters/music/scanner/scanner.go:172]:
```suggestion
			return fmt.Errorf("loop detected: '%s' visited multiple times", realPath)
```

File: adapters/music/scanner/scanner_test.go [hunk 2]
```diff
@@ -296,6 +345,11 @@ func TestScanner(t *testing.T) {
 				Tracks: map[string]scanner.Track{},
 			},
 		},
+		{
+			Name:   "symlinks_loop",
+			Result: scanner.Album{},
+			Error:  errors.New("initial load failed: walk failed: look detected: 'test_data/symlinks_loop/a' visited multiple times"),
```
Comment (boreq) [inline adapters/music/scanner/scanner_test.go:351]:
```suggestion
			Error:  errors.New("initial load failed: walk failed: loop detected: 'test_data/symlinks_loop/a' visited multiple times"),
```

**Pull request merged**